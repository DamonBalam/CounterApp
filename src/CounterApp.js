import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ valor = 10 }) => {
    const [counter, setCounter] = useState(valor);

    const handleClick = () => {
        setCounter(counter + 1);
    };

    const handleSubtract = () => {
        setCounter(valor);
    };

    const handleReset = () => {
        setCounter(counter - 1);
    };

    return (
        <Fragment>
            <h2>CounterApp</h2>
            <h3> {counter}</h3>
            <button onClick={handleClick}>+1</button>
            <button onClick={handleSubtract}>Reset</button>
            <button onClick={handleReset}> -1</button>
        </Fragment>
    );
};

CounterApp.propTypes = {
    valor: PropTypes.number,
};

export default CounterApp;
