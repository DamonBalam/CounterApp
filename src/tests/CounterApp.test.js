import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import CounterApp from '../CounterApp';
describe('Pruebas de CounterApp', () => {
    let wrapper = shallow(<CounterApp />);

    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    });

    test('Debe mostrar <CounterApp/> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar el valor por defecto 100', () => {
        const valor = 100;
        const wrapper = shallow(<CounterApp valor={valor} />);

        const texth3 = wrapper.find('h3').text().trim();

        expect(texth3).toBe(`${valor}`);
    });

    test('Debe incrementar contador +1', () => {
        wrapper.find('button').at(0).simulate('click');

        const texth3 = wrapper.find('h3').text().trim();

        expect(texth3).toBe('11');
    });

    test('Debe decrementar contador -1', () => {
        wrapper.find('button').at(2).simulate('click');

        const texth3 = wrapper.find('h3').text().trim();

        expect(texth3).toBe('9');
    });

    test('Debe resetear el valor por defecto con el boton de reset', () => {
        
        const wrapper = shallow(<CounterApp valor={105} />);

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');

        wrapper.find('button').at(1).simulate('click');

        const texth3 = wrapper.find('h3').text().trim();

        expect(texth3).toBe('105');
    });
});
