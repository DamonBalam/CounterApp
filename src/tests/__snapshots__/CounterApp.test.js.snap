// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de CounterApp Debe mostrar <CounterApp/> correctamente 1`] = `
<Fragment>
  <h2>
    CounterApp
  </h2>
  <h3>
     
    10
  </h3>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
     -1
  </button>
</Fragment>
`;
